nsxt/data_source_nsxt_policy_host_transport_node_collection_state.go

```diff
@@ -0,0 +1,132 @@
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+var policyHostTransportNodeCollectionState = []string{
+	model.TransportNodeCollectionState_STATE_FAILED_TO_CREATE,
+	model.TransportNodeCollectionState_STATE_FAILED_TO_REALIZE,
+	model.TransportNodeCollectionState_STATE_IN_PROGRESS,
+	model.TransportNodeCollectionState_STATE_PROFILE_MISMATCH,
+	model.TransportNodeCollectionState_STATE_SUCCESS,
+}
+
+func dataSourceNsxtPolicyHostTransportNodeCollectionState() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNsxtPolicyHostTransportNodeCollectionStateRead,
+
+		Schema: map[string]*schema.Schema{
+			"id":           getDataSourceIDSchema(),
+			"display_name": getDataSourceDisplayNameSchema(),
+			"description":  getDataSourceDescriptionSchema(),
+			"path":         getPathSchema(),
+			"site_path": {
+				Type:         schema.TypeString,
+				Description:  "Path of the site this Transport Node Collection belongs to",
+				Optional:     true,
+				ValidateFunc: validatePolicyPath(),
+			},
+			"aggregate_progress_percentage": {
+				Type:        schema.TypeInt,
+				Description: "Aggregate percentage of compute collection deployment",
+				Computed:    true,
+			},
+			"cluster_level_error": {
+				Type:        schema.TypeString,
+				Description: "Errors which needs cluster level to resolution",
+				Optional:    true,
+				Computed:    true,
+			},
+			"state": {
+				Type:         schema.TypeString,
+				Description:  "Application state of transport node profile on compute collection",
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice(policyHostTransportNodeCollectionState, false),
+			},
+			"validation_errors": {
+				Type:        schema.TypeList,
+				Description: "Errors while applying transport node profile on discovered node",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"discovered_node_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"error_message": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+				Optional: true,
+				Computed: true,
+			},
+			"vlcm_transition_error": {
+				Type:        schema.TypeString,
+				Description: "Errors while enabling vLCM on the compute collection",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceNsxtPolicyHostTransportNodeCollectionStateRead(d *schema.ResourceData, m interface{}) error {
+	if isPolicyGlobalManager(m) {
+		return localManagerOnlyError()
+	}
+
+	objID := d.Get("id").(string)
+	objName := d.Get("display_name").(string)
+	objSitePath := d.Get("site_path").(string)
+	// For local manager, if site path is not provided, use default site
+	if objSitePath == "" {
+		objSitePath = defaultSite
+	}
+	if objID == "" && objName == "" {
+		return fmt.Errorf("please specify id or display_name in order to identify Transport Node Collection")
```

Comment:
Wouldn't `site_path` be enough to identify transport node collection, in case there is single collection per site? In this case, I would suggest to allow not to specify name or id.

Reply:
Removed the restraint.